### ISSUE TITLE
[BUG] Checkpointing doesn't work for FBD trees w/ nonzero offset from the present

### ISSUE THREAD

**Issue by davidcerny** [revbayes/revbayes#531]
**Describe the bug**
Since #484, `BirthDeathSamplingTreatmentProcess::setValue()` has called `TreeUtilities::startingTreeInitializer()` to restore fossil age ranges when restarting FBD analyses from a checkpoint. Unfortunately, this solution only works for trees with extant tips. If a tree is entirely extinct (i.e., has a nonzero offset from the present), the restart attempt will fail, since the youngest tip of the tree stored in the checkpoint file is assumed to be of age 0. As a result, the tree will fail the check performed by `startingTreeInitializer()`, because the ages of its tips fall outside of specified bounds.

**To reproduce**
Run the Rev script from [this ZIP archive](https://github.com/user-attachments/files/16615970/fbd_test.zip). One absolute path has to be changed. 

**Expected behavior**
We should be able to restart FBD analyses from a checkpoint file without failing the check.

**Computer info**
macOS 14.5 (Sonoma), ARM64 architecture, RevBayes 1.2.5-preview, latest version of the `development` branch (f2a0610).

**Attempted solution**
In principle, the value of the offset _is_ recoverable, and we could modify `setValue()` by inserting something like this before the call to `startingTreeInitializer()`:

```c_cpp
for (size_t i = 0; i < taxa.size(); i++)
{
    double new_age = v->getTipNodeWithName( taxa[i].getName() ).getAge() + some_offset;
    v->getTipNodeWithName( taxa[i].getName() ).setAge( new_age );
}
```

The problem is figuring out the value of `some_offset`. It's equal to the age of the youngest tip sampled in the checkpointed iteration, but tip ages are only stored in the `.log` file rather than the `.state` file. The `.state` file does, however, store both the tree (the `fbd_tree` column in the example above) and the process age (the `origin_time` column in the example above; it can be either the age of the origin or of the root, depending on how BDSTP is set up). So the offset can also be computed by subtracting the root age of the stored tree (plus the length of the root edge if there is one) from the process age. I tried the following:

```
some_offset = getOriginAge() - v->getRoot().getAge();
```

but that doesn't work; `getOriginAge()` doesn't return the value stored in the checkpoint file (approximately 250.756 in the example above) but just a random draw from the prior. BDSTP also has a private member called `offset`, but that one is simply initialized from the `taxa` vector: in the example above, it is set to 66.2, while the true offset is 66.05169.

**Tl;dr:** We need to extract values from two different columns of the `.state` file: not just the one that stores the tree but also the one that stores the process age.

**Comment by davidcerny** [comment]
Hmm... so my proposed solution won't work. Even with the correct offset hardcoded into `setValue()`, the hanging problem from issue #516 is back with a vengeance.

**Comment by davidcerny** [comment]
(Note to self: the hanging problem goes away if we increment the ages of _all_ nodes by the offset, and not just the tips.)
